### PR TITLE
[chore] Use business application hosted in the repository for the Apache HTTPD instrumentation E2E test

### PR DESCRIPTION
**Pull request by iblancasa**
Related to #2073.

Other tests will be migrated in follow up PRs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -202,8 +208,13 @@ e2e-log-operator:
...
 
 .PHONY: prepare-e2e
-prepare-e2e: kuttl set-image-controller container container-target-allocator container-operator-opamp-bridge start-kind cert-manager install-metrics-server load-image-all deploy
-	TARGETALLOCATOR_IMG=$(TARGETALLOCATOR_IMG) SED_BIN="$(SED)" ./hack/modify-test-images.sh
+prepare-e2e: kuttl set-image-controller container container-target-allocator container-operator-opamp-bridge start-kind cert-manager install-metrics-server load-image-all deploy instrumentation-e2e-apps
+	APACHE_E2E=$(IMG_PREFIX)/apachehttpd-test:e2e TARGETALLOCATOR_IMG=$(TARGETALLOCATOR_IMG) SED_BIN="$(SED)" ./hack/modify-test-images.sh
+
+.PHONY: build-instrumentation-e2e-apps
+instrumentation-e2e-apps:
+	docker buildx build --load --platform linux/${ARCH} -t $(IMG_PREFIX)/apachehttpd-test:e2e tests/instrumentation-e2e-apps/apache-httpd/apache-httpd
```
Comment (iblancasa) [inline Makefile:216]: This path is not correct

File: Makefile [hunk 2]
```diff
@@ -202,8 +208,13 @@ e2e-log-operator:
 	kubectl get deploy -A
 
 .PHONY: prepare-e2e
-prepare-e2e: kuttl set-image-controller container container-target-allocator container-operator-opamp-bridge start-kind cert-manager install-metrics-server load-image-all deploy
-	TARGETALLOCATOR_IMG=$(TARGETALLOCATOR_IMG) SED_BIN="$(SED)" ./hack/modify-test-images.sh
+prepare-e2e: kuttl set-image-controller container container-target-allocator container-operator-opamp-bridge start-kind cert-manager install-metrics-server load-image-all deploy instrumentation-e2e-apps
+	APACHE_E2E=$(IMG_PREFIX)/apachehttpd-test:e2e TARGETALLOCATOR_IMG=$(TARGETALLOCATOR_IMG) SED_BIN="$(SED)" ./hack/modify-test-images.sh
+
+.PHONY: build-instrumentation-e2e-apps
```
Comment (iblancasa) [inline Makefile:214]: Fix this

**Comment by pavolloffay** [comment]
Please split the PR:
* instrumentation test suite
* add httpd example app

also I don't see GHA workflow to publish the app image

**Comment by iblancasa** [comment]
> Please split the PR:
> 
> * instrumentation test suite
> * add httpd example app
> 
> also I don't see GHA workflow to publish the app image

I'll create new PRs for that.

**Pull request closed without merging**